test/integration/supervisor_login_test.go

```diff
@@ -128,6 +131,14 @@ func TestSupervisorLogin(t *testing.T) {
 			wantDownstreamIDTokenSubjectToMatch:  "^" + regexp.QuoteMeta(env.SupervisorUpstreamOIDC.Issuer+"?sub=") + ".+",
 			wantDownstreamIDTokenUsernameToMatch: func(_ string) string { return "^" + regexp.QuoteMeta(env.SupervisorUpstreamOIDC.Username) + "$" },
 			wantDownstreamIDTokenGroups:          env.SupervisorUpstreamOIDC.ExpectedGroups,
+			editRefreshSessionDataWithoutBreaking: func(t *testing.T, sessionData *psession.PinnipedSession) {
+				// even if we update this group to the wrong thing, we expect that it will return to the correct
+				// value after we refresh.
+				// However if there are no expected groups then they will not update, so we should skip this.
+				if len(env.SupervisorUpstreamOIDC.ExpectedGroups) > 0 {
```

Comment:
In OIDC, if we don't get a groups claim back, we don't update the previous value (it could mean that we don't get an id token back and we don't have a user info endpoint, which is a valid configuration as long as the refresh token can still be exchanged).
In the case of dex, we don't have groups at all because the local user store is a minimal testing setup, so we will never get groups back upon refresh. This means that this test doesn't really make sense for that case.